cmd/ebrelayer/relayer/ethereum.go

```diff
@@ -271,9 +280,9 @@ func (sub EthereumSub) Start(completionEvent *sync.WaitGroup) {
 				}
 				time.Sleep(transactionInterval)
 			}
-			// save the current ending block to the lastprocessed block to ensure we keep reading blocks sequentially
-			lastProcessedBlock = endingBlock
-			err = db.Put([]byte(ethLevelDBKey), lastProcessedBlock.Bytes(), nil)
+
+			// save the current ending block + 1 to the lastprocessed block to ensure we keep reading blocks sequentially and don't repeat blocks
+			err = db.Put([]byte(ethLevelDBKey), endingBlock.Add(endingBlock, big.NewInt(1)).Bytes(), nil)
```

Comment:
can we move endingBlock.Add(endingBlock, big.NewInt(1)) onto its own line?  It's easy to miss that the call to db.Put is also changing the value of endingBlock.

Reply:
Sure.